File: tlg/src/ch/bailu/tlg/ByteInteger.java
package ch.bailu.tlg;

import java.io.BufferedInputStream;
import java.io.BufferedOutputStream;
import java.io.IOException;
import java.nio.ByteBuffer;

public class ByteInteger {
    private byte [] buffer=new byte[4];

    public static ByteInteger wrap(int i) {
        return new ByteInteger(i);
    }
    
    public static int read(BufferedInputStream reader) throws IOException {
        return new ByteInteger(reader).getValue();
    }
    
    public ByteInteger(int i) {
        ByteBuffer wrapper =ByteBuffer.wrap(buffer);
        wrapper.putInt(i);
    }
    
    public ByteInteger(BufferedInputStream reader) throws IOException {
        reader.read(buffer);
    }

    public int getValue() {
        return ByteBuffer.wrap(buffer).getInt();
    }
    
    public void writeState(BufferedOutputStream output) throws IOException {
        output.write(buffer);
    }
    
    
}

File: tlg/src/main/kotlin/context/ByteInteger.kt
package context

import java.io.BufferedInputStream
import java.io.BufferedOutputStream
import java.io.IOException
import java.nio.ByteBuffer


class ByteInteger {
    private val buffer = ByteArray(4)

    constructor(i: Int) {
        val wrapper = ByteBuffer.wrap(buffer)
        wrapper.putInt(i)
    }

    constructor(reader: BufferedInputStream) {
        reader.read(buffer)
    }

    val value: Int
        get() = ByteBuffer.wrap(buffer).int

    @Throws(IOException::class)
    fun writeState(output: BufferedOutputStream) {
        output.write(buffer)
    }

    companion object {
        fun wrap(i: Int): ByteInteger {
            return ByteInteger(i)
        }

        @Throws(IOException::class)
        fun read(reader: BufferedInputStream): Int {
            return ByteInteger(reader).value
        }
    }
}
